### ISSUE TITLE
backfill: lazy backfill

### ISSUE THREAD

**Issue by annrpom** [cockroachdb/cockroach#126850]
We should look into doing something like https://www.2ndquadrant.com/en/blog/add-new-table-column-default-value-postgresql-11/:

> In a new feature I worked on and committed, the default value is just stored in the catalog and used where needed in rows existing at the time the change was made. New rows, and new versions of existing rows, are written with the default value in place, as happens now.

We might consider allowing users to specify when to issue the rest of the backfill job to avoid periods of heavy cluster traffic

https://cockroachlabs.slack.com/archives/C04N0AS14CT/p1716479612947039

**Comment by annrpom** [comment]
Hi @XiaochenCui: more product decisions need to be ironed out before we can consider implementing this. Feel free to prototype this if you are interested, though. During prototyping:
1. feel free to ignore implementing this in our "[declarative schema changer](https://github.com/cockroachdb/cockroach/blob/c063b43f7e082bf8869992d651b0be4dbac83067/pkg/sql/schemachanger/doc.go#L13-L18)" (aka you can avoid everything in the `schemachanger` directory)
2. this should be an opt-in feature and should be enabled by a cluster setting. feel free to refer to other PRs for how to implement a cluster setting; i'll just throw [this](https://github.com/cockroachdb/cockroach/pull/126860) in as an example. 
```
// In general,
// this is how you create a cluster setting:
var yourNiceNameHere = settings.RegisterBoolSetting(
	settings.ApplicationLevel,
	"sql.backfill.lazy_backfill.enabled",
	"determines bleh",
	false,
	settings.WithPublic,
)
...
// this is how you use:
if yourNiceNameHere.Get(&p.ExecCfg().Settings.SV) {
}
```

**Comment by rafiss** [comment]
closing in favor of https://github.com/cockroachdb/cockroach/issues/98415
